src/osdp_cp.c

```diff
@@ -1479,6 +1494,14 @@ osdp_t *osdp_cp_setup(int num_pd, const osdp_pd_info_t *info_list)
 	return (osdp_t *)__cp_setup(num_pd, info_list);
 }
 
+int osdp_cp_add_pd(osdp_t *ctx, int num_pd, const osdp_pd_info_t *info)
+{
+	assert(ctx);
```

Comment:
Should be `input_check(ctx);` as `osdp_cp_add_pd()` must be called _only_ after a successful call to `osdp_cp_setup()`

```suggestion
	input_check(ctx);
```

Reply:
done